app/src/main/java/com/example/concordia_campus_guide/Models/Place.java

```diff
@@ -1,20 +1,21 @@
 package com.example.concordia_campus_guide.Models;
 
-import com.example.concordia_campus_guide.Database.Converters.DoubleListConverter;
-import com.example.concordia_campus_guide.Database.Converters.StringListConverter;
-import com.google.android.gms.maps.model.LatLng;
-
 import androidx.room.ColumnInfo;
 import androidx.room.Entity;
-import androidx.room.Ignore;
+import androidx.room.Index;
 import androidx.room.PrimaryKey;
-import androidx.room.TypeConverter;
 import androidx.room.TypeConverters;
 
-@Entity(tableName = "places")
-public class Place {
+import com.example.concordia_campus_guide.Database.Converters.DoubleListConverter;
+import com.google.android.gms.maps.model.LatLng;
+
+@Entity(tableName = "places",
+        indices = {@Index("id")}
+        )
+public abstract class Place {
 
-    @Ignore
+    @ColumnInfo(name = "id")
```

Comment:
Ok, so this will be considered as the primary key in the buildings table instead right ?

Reply:
yes, room, floor, building all use the id as pk now.

since their previous pk (building_code, floor_code, room_code) are unique, we can still use them for our relations